taichi/codegen/spirv/spirv_ir_builder.cpp

```diff
@@ -392,6 +392,14 @@ SType IRBuilder::get_pointer_type(const SType &value_type,
   return t;
 }
 
+SType IRBuilder::get_underlying_image_type(const SType &primitive_type,
+                                           int num_dimensions) {
```

Comment:
I would like to instead of getting a value from a map, have this function construct the image type. Essentially moving this code to here:
```c++
int img_id = id_counter_++;
  spv::Dim dim;
  if (num_dimensions == 1) {
    dim = spv::Dim1D;
  } else if (num_dimensions == 2) {
    dim = spv::Dim2D;
  } else if (num_dimensions == 3) {
    dim = spv::Dim3D;
  } else {
    TI_ERROR("Unsupported number of dimensions: {}", num_dimensions);
  }
  ib_.begin(spv::OpTypeImage)
      .add_seq(img_id, primitive_type, dim,
               /*Depth=*/0, /*Arrayed=*/0, /*MS=*/0, /*Sampled=*/1,
               spv::ImageFormatUnknown)
      .commit(&global_);
```